spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/mail/MailSenderAutoConfiguration.java

```diff
@@ -40,6 +41,7 @@
  */
 @Configuration
 @ConditionalOnClass({MailSender.class, AmazonSimpleEmailService.class})
+@ConditionalOnMissingBean(MailSender.class)
```

Comment:
I think these autoconfigurers are intended to provide sensible defaults that users can override. Assuming that's true, it's surprising when a user configures a `MailSender` bean and then `org.springframework.cloud.aws.autoconfigure.mail.MailSenderAutoConfiguration` overrides it with no (obvious) way to stop it from doing so.

I also think the purpose of the autoconfigurers is to serve the more common cases and not solve for everything. In your example, having mail senders configured for both Google and AWS at the same time is (I think, at least) highly unusual - an application almost always has exactly 1 mail sender. For those users who want multiple, they can certainly do so, but not using the autoconfigurers.